cmake/FindPoly.cmake

```diff
@@ -143,6 +143,14 @@ if(NOT Poly_FOUND_SYSTEM)
       "${DEPS_BASE}/lib/libpicpolyxx${CMAKE_STATIC_LIBRARY_SUFFIX}")
   endif()
 
+  set(POLY_CC_FLAGS )
+  if(NOT(WASM STREQUAL "OFF"))
+    set(POLY_CC_FLAGS 
+          -DCMAKE_CXX_FLAGS=-Wno-error=unused-but-set-variable
+          -DCMAKE_C_FLAGS=-Wno-error=unused-but-set-variable
+        )
```

Comment:
Can you add a comment that describes why this warning suppression is required (with maybe a link to the affected code)? This will make it easier to remove this special case in the future when it is not needed anymore.